proxy/StatPages.cc

```diff
@@ -83,7 +83,7 @@ StatPagesManager::handle_http(Continuation *cont, HTTPHdr *header)
     host_len       = unescapifyStr(host);
 
     for (i = 0; i < n_stat_pages; i++) {
-      if (ptr_len_cmp(host, host_len, stat_pages[i].module) == 0) {
+      if ((host_len == sizeof(stat_pages[i].module)) && (memcmp(host, stat_pages[i].module, host_len) == 0)) { 
```

Comment:
Looks like that should work and we won't need the len checks. According to the doc...

This function starts comparing the first character of each string. If they are equal to each other, it continues with the following pairs until the characters differ, until a terminating null-character is reached, or until num characters match in both strings, whichever happens first.